contracts/dmarket.js

```diff
@@ -0,0 +1,1285 @@
+/* eslint-disable no-await-in-loop */
+/* global actions, api */
+
+const UTILITY_TOKEN_SYMBOL = 'BEE';
+
+// 30 days in seconds
+const MAX_EXPIRATION_SECS = 2592000;
+// fee in UTILITY TOKEN required to create a new pair
+const PAIR_CREATION_FEE = '500';
+
+const CONTRACT_NAME = 'dmarket';
+
+actions.createSSC = async () => {
+  const tableExists = await api.db.tableExists('pairs');
+
+  if (tableExists === false) {
+    await api.db.createTable('pairs', ['symbol', 'precision', 'allowedSymbols']);
+    await api.db.createTable('buyBook', ['symbol', 'pair', 'account', 'priceDec', 'expiration', 'txId']);
+    await api.db.createTable('sellBook', ['symbol', 'pair', 'account', 'priceDec', 'expiration', 'txId']);
+    await api.db.createTable('tradesHistory', ['symbol', 'pair']);
+    await api.db.createTable('metrics', ['symbol', 'pair']);
+
+    // default global pairs
+    const pairs = [
+      {
+        pair: 'BEE',
+        precision: 8,
+        allowedSymbols: true, // true - all tokens are allowed
+      },
+      {
+        pair: 'SWAP.BTC',
+        precision: 8,
+        allowedSymbols: true, // true - all tokens are allowed
```

Comment:
Suggestion here is instead of re-using allowedSymbols, we have a separate indicator 'global' to check if present and use that instead.

Reply:
In fact, maybe we should change the action to be 'setGlobalPair', so that it can be toggled on (but probably should not be allowed to turn it off)

Reply:
added a separate indicator 'isGlobal'